misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -79,4 +87,82 @@ static void sortImportDeclarations(List<ImportDeclarationNode> importDeclaration
+        for (int i = 0; i < importNodes.size(); i++) {
+            if (isImportsEqual(firstImportOrgName, firstImportModuleName, importNodes.get(i))) {
+                prevFirstIndex = i;
+                break;
+            }
+        }
+        if (prevFirstIndex == 0) {
+            return;
+        }
+        // remove comments from the previous first import
+        ImportDeclarationNode prevFirstImportNode = importNodes.get(prevFirstIndex);
+        MinutiaeList prevLeadingMinutiae = prevFirstImportNode.leadingMinutiae();
+        List<Minutiae> leadingMinutiae = new ArrayList<>();
+        Minutiae prevFirstMinutiae = prevLeadingMinutiae.get(0);
+        if (prevFirstMinutiae.kind() != SyntaxKind.COMMENT_MINUTIAE) {
+            leadingMinutiae.add(prevFirstMinutiae);
+            prevLeadingMinutiae = prevLeadingMinutiae.remove(0);
+        }
+
+        Token prevFirstImportToken = prevFirstImportNode.importKeyword();
+        Token modifiedPrevFirstImportToken =
+                prevFirstImportToken.modify(NodeFactory.createMinutiaeList(leadingMinutiae),
+                        prevFirstImportToken.trailingMinutiae());
+        importNodes.set(prevFirstIndex,
+                prevFirstImportNode.modify().withImportKeyword(modifiedPrevFirstImportToken).apply());
+
+        // add leading comments from the previous first import
+        if (!hasEmptyLineAtEnd(prevLeadingMinutiae)) {
+            prevLeadingMinutiae =
+                    prevLeadingMinutiae.add(NodeFactory.createEndOfLineMinutiae(System.lineSeparator()));
+        }
+        ImportDeclarationNode sortedFirstImportNode = importNodes.get(0);
+        Token sortedFirstImportToken = sortedFirstImportNode.importKeyword();
+        MinutiaeList sortedLeadingMinutiae = sortedFirstImportToken.leadingMinutiae();
+        for (int i = 0; i < sortedLeadingMinutiae.size(); i++) {
+            Minutiae minutiae = sortedLeadingMinutiae.get(i);
+            if (i == 0 && minutiae.kind() == SyntaxKind.END_OF_LINE_MINUTIAE) {
+                continue;
+            }
+            prevLeadingMinutiae = prevLeadingMinutiae.add(minutiae);
+        }
+        Token modifiedSortedFirstImportToken =
+                sortedFirstImportToken.modify(prevLeadingMinutiae, sortedFirstImportToken.trailingMinutiae());
+        importNodes.set(0,
+                sortedFirstImportNode.modify().withImportKeyword(modifiedSortedFirstImportToken).apply());
+    }
+
+    private static boolean isImportsEqual(String firstImportOrgName, String firstImportModuleName,
+                                          ImportDeclarationNode secondImportNode) {
+        Optional<ImportOrgNameNode> orgNameNode = secondImportNode.orgName();
+        String importOrgName = orgNameNode.isPresent() ? orgNameNode.get().toSourceCode() : "";
+        if (firstImportOrgName.equals(importOrgName)) {
+            return firstImportModuleName.equals(getImportModuleName(secondImportNode));
+        }
+        return false;
+    }
```

Comment:
```suggestion
    private static boolean doesImportMatch(String orgName, String moduleName, ImportDeclarationNode importDeclNode) {
        String importDeclOrgName = importDeclNode.orgName().map(ImportOrgNameNode::toSourceCode).orElse("");
        String importDeclModuleName = getImportModuleName(importDeclNode);
        return orgName.equals(importDeclOrgName) && moduleName.equals(importDeclModuleName);
    }
```

Reply:
As per this comment, we don't need to compare the moduleName when the orgNames are different right?. https://github.com/ballerina-platform/ballerina-lang/pull/41481#discussion_r1547497476. So we can avoid the extra call to getImportModuleName function. So modified the last check to this. 
`return orgName.equals(importDeclOrgName) && moduleName.equals(getImportModuleName(importDeclNode));`